bcipy/simulator/helpers/signal_helpers.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging as logger
+from dataclasses import dataclass
+from typing import Tuple
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from bcipy.config import (
+    RAW_DATA_FILENAME,
+    TRIGGER_FILENAME,
+    DEFAULT_PARAMETER_FILENAME, SESSION_DATA_FILENAME,
+    DEFAULT_DEVICE_SPEC_FILENAME,
+)
+from bcipy.helpers.acquisition import analysis_channels
+import bcipy.acquisition.devices as devices
+from bcipy.helpers.list import grouper
+from bcipy.helpers.load import load_json_parameters, load_raw_data, load_experimental_data
+from bcipy.helpers.session import read_session, evidence_records
+from bcipy.helpers.stimuli import update_inquiry_timing
+from bcipy.helpers.triggers import TriggerType, trigger_decoder
+from bcipy.helpers.symbols import alphabet
+from bcipy.signal.model import PcaRdaKdeModel
+from bcipy.signal.process import get_default_transform, filter_inquiries, ERPTransformParams
+
+logger.getLogger().setLevel(logger.INFO)
+log = logger.getLogger(__name__)
+
+
+@dataclass()
+class ExtractedExperimentData: # TODO clean up design
+    inquiries: np.ndarray
+    trials: np.ndarray
+    labels: list
+    inquiry_timing: list
+
+    decoded_triggers: tuple
+
+
+def process_raw_data_for_model(data_folder, parameters, model_class=PcaRdaKdeModel) -> ExtractedExperimentData:
```

Comment:
If you look at Basak's latest review you can see that each model is probably going to have different data processing needs. You may want to think of some ways to match up a model with it data processor.